annotations/Makefile

```diff
@@ -25,17 +25,25 @@ DIST_JAR = $(DIST_JAR_DIR)/$(JAR_NAME)
 
 CLASSES_DIR = $(BUILD_DIR)/classes
 JAVA_SOURCE_DIR = src/main/java
+JAVA9_CLASSES_DIR = $(BUILD_DIR)/classes-java9
+JPMS_CLASS = $(JAVA9_CLASSES_DIR)/module-info.class
```

Comment:
Classes for Java 9 are built together with the `module-info.class`. This is necessary because the `annotations` should export all annotation classes, and `javac` wants to be able to see those during the module build.

We don't want to include the duplicate Java 9 bytecode classes, though, just the `module-info.class`. So they are _built_ together in this dedicated directory, but only the `module-info.class` makes it into the final MR JAR.

Reply:
SGTM